feat: add support for Nutri-Patrol API

bengeois opened the pull request:
### What
Nutripatrol API wrapper

Fix https://github.com/openfoodfacts/openfoodfacts-nodejs/issues/520

bengeois committed:
feat: add nutripatrol API interfaces and support all flag endpoints

bengeois committed:
feat: add support for ticket in nutripatrol API

bengeois committed:
Merge branch 'develop' into feature/nutripatrol

VaiTon commented:
Hey @bengeois, thanks for your work! We don't usually throw errors. Instead we return them as a Union. Can we please also do that in this section of the API?

See https://github.com/openfoodfacts/openfoodfacts-nodejs/blob/71b84960bd96da541d02470a0ecfa35b01cfb907/src/folksonomy.ts#L112 for reference.

bengeois committed:
Merge branch 'openfoodfacts:develop' into feature/nutripatrol

bengeois committed:
feat: refactor error handling in nutripatrol wrapper

bengeois commented:
Hey @VaiTon , thanks for your feedback!

Initially, I was using catch blocks to separate error handling logic from the standard flow. However, I've refactored the code as requested to align with the existing approach used in other wrappers, where errors are returned as a Union.

Additionally, I’ve introduced a NutriPatrolError to provide more detailed information, such as the statusCode. This allows us to maintain useful error details that would otherwise be lost if we didn't include them.

Let me know if you have any other thoughts!

teolemon commented:
Thanks @bengeois and congratulations on this first feature 👌
